Amethyst/Managers/ScreenManager.swift

```diff
@@ -125,8 +133,14 @@ final class ScreenManager: NSObject {
 
         let windows = (delegate?.activeWindowsForScreenManager(self) ?? [])
         changingSpace = false
-        reflowOperation = currentLayout?.reflow(windows, on: screen)
-        OperationQueue.main.addOperation(reflowOperation!)
+
+        let reflowOperation = currentLayout?.reflow(windows, on: screen)
+        reflowOperation?.onReflowCompletion = { [unowned self] in
+            // propagate the completion handler if we have one
+            self.onReflowCompletion?()
+        }
+        self.onReflowInitiation?()
+        reflowOperation?.enqueue(OperationQueue.main)
```

Comment:
Out of curiosity, is the function here to avoid having to unwrap the `reflowOperation`? If so it would probably be better to just do the conditional unwrap with
```swift
if let reflowOperation = reflowOperation {
    ...
}
```